projects/js-packages/publicize-components/src/components/share-buttons/share-buttons.tsx

```diff
@@ -52,19 +53,31 @@ export function ShareButtons( { buttonStyle = 'icon', buttonVariant }: ShareButt
 	);
 
 	return (
-		<div className={ styles[ 'share-buttons' ] }>
+		<div
+			className={ classnames(
+				styles[ 'share-buttons' ],
+				// If we are showing the text, we will show the buttons vertically.
+				{ [ styles.vertical ]: buttonStyle.includes( 'text' ) }
+			) }
+		>
 			{ availableNetworks.map( ( { label, networkName, url } ) => {
 				const href = prepareText( url );
 
 				const icon =
 					'icon' === buttonStyle ? <SocialServiceIcon serviceName={ networkName } /> : null;
 
+				const text = sprintf(
+					/* translators: %s is the name of a social network, e.g. Twitter. */
+					__( 'Share on %s', 'jetpack' ),
+					label
+				);
+
 				return (
 					<div className={ styles.container } key={ networkName }>
 						<Button
 							icon={ icon }
 							variant={ buttonVariant }
-							aria-label={ label }
+							aria-label={ text }
```

Comment:
This is definitely better than just having the network name here, but could this cause problems with the Text component below and cause the label to be read twice? Is there a way to more closely associate the two components? Or perhaps use aria-hidden on the text?

Reply:
> aria-label takes precedence over the text included in the button

https://www.w3.org/WAI/WCAG22/Techniques/general/G208.html